instrumentation/rabbitmq-2.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/RabbitMqTest.java

```diff
@@ -0,0 +1,955 @@
+                          false,
+                          false);
+                    }));
+  }
+
+  @Test
+  void testSpringRabbit() {
+    CachingConnectionFactory cachingConnectionFactory =
+        new CachingConnectionFactory(connectionFactory);
+    AmqpAdmin admin = new RabbitAdmin(cachingConnectionFactory);
+    AmqpTemplate template = new RabbitTemplate(cachingConnectionFactory);
+
+    Queue queue = new Queue("some-routing-queue", false, true, true, null);
+    testing.runWithSpan(
+        "producer parent",
+        () -> {
+          admin.declareQueue(queue);
+          template.convertAndSend(queue.getName(), "foo");
+        });
+
+    String message =
+        testing.runWithSpan(
+            "consumer parent", () -> (String) template.receiveAndConvert(queue.getName()));
+
+    assertEquals("foo", message);
+
+    AtomicReference<SpanData> producerSpan = new AtomicReference<>();
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace
+                .hasSize(3)
+                .hasSpansSatisfyingExactly(
+                    span -> {
+                      span.hasName("producer parent").hasKind(SpanKind.INTERNAL).hasNoParent();
+                    },
+                    span -> {
+                      rabbitSpan(
+                          trace, span, 1, null, null, null, "queue.declare", trace.getSpan(0));
+                    },
+                    span -> {
+                      rabbitSpan(
+                          trace,
+                          span,
+                          2,
+                          "<default>",
+                          "some-routing-queue",
+                          "publish",
+                          "<default>",
+                          trace.getSpan(0));
+                      producerSpan.set(trace.getSpan(2));
+                    }),
+        trace -> {
+          trace
+              .hasSize(2)
+              .hasSpansSatisfyingExactly(
+                  span -> {
+                    span.hasName("consumer parent").hasKind(SpanKind.INTERNAL).hasNoParent();
+                  },
+                  span -> {
+                    rabbitSpan(
+                        trace,
+                        span,
+                        1,
+                        "<default>",
+                        "some-routing-queue",
+                        "receive",
+                        queue.getName(),
+                        trace.getSpan(0),
+                        producerSpan.get(),
+                        null,
+                        null,
+                        false,
+                        false);
+                  });
+        });
+
+    cachingConnectionFactory.destroy();
+  }
+
+  @Test
+  void testMessageHeaderAsSpanAttributes() throws IOException, InterruptedException {
+    String queueName = channel.queueDeclare().getQueue();
+    Map<String, Object> headers = new java.util.HashMap<>();
+    headers.put("test-message-header", "test");
```

Comment:
It was already confusing even in groovy (you inherited this), but it's not super apparent where this header is asserted. It's buried down in the `rabbitSpan` method and a boolean argument tells it to check a hard-coded value for `test`. 🤦🏻

Reply:
Agreed it's messy. As it's only for one test I will move it out of `rabbitSpan` and into the test method itself